### PR TITLE
Support MaxMessageBatchSize on trigger

### PR DESCRIPTION
**Pull request by apranaseth**
Support MaxMessageBatchSize on trigger. Use it from trigger when its set, instead from host.json.
This allows different triggers with different batch size requirements to be kept in the same app.

### CODE REVIEW

**Comment by jsquire** [comment]
@JoshLove-msft:  Would you take point for this one?

**Comment by jsquire** [comment]
@apranaseth:  We'll need to add a change log entry for this, and also to update the MS docs content for the `host.json` file to call out that the value may be overridden locally.

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs [hunk 1]
```diff
@@ -168,7 +171,7 @@ internal sealed class ServiceBusListener : IListener, IScaleMonitorProvider, ITa
 
             if (_supportMinBatchSize)
             {
-                _cachedMessagesManager = new(options.MaxMessageBatchSize, options.MinMessageBatchSize);
+                _cachedMessagesManager = new(_maxMessageBatchSize, options.MinMessageBatchSize);
```
Comment (m-redding) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs:174]: I think we will still need to support setting max message batch size from the host.json, but just prefer the value set in the trigger.

i.e.:

```suggestion
                _cachedMessagesManager = new(_maxMessageBatchSize > 0 ? _maxMessageBatchSize : options.MaxMessageBatchSize, options.MinMessageBatchSize);
```

Comment (m-redding) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs:174]: Nevermind I see you're already handling that upstream - my bad!

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/ServiceBusTriggerAttribute.cs [hunk 2]
```diff
@@ -106,12 +107,34 @@ public bool AutoCompleteMessages
...
 
+        /// <summary>
+        /// Gets or sets the maximum number of messages that will be passed to each function call. This only applies for functions that receive
+        /// a batch of messages.
+        /// </summary>
+        public int MaxMessageBatchSize
+        {
+            get
+            {
+                return _maxMessageBatchSize.HasValue ? _maxMessageBatchSize.Value : 0;
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/ServiceBusTriggerAttribute.cs:118]: Should this use the same default that we define in the host.json setting? https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusOptions.cs#L178

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/Bindings/ServiceBusTriggerAttributeBindingProviderTests.cs [hunk 3]
```diff
@@ -108,6 +108,43 @@ public async Task GetAutoCompleteMessagesOptionToUse_AutoCompleteDisabledOnTrigg
...
+            var context = new TriggerBindingProviderContext(parameter, CancellationToken.None);
+            var binding = await _provider.TryCreateAsync(context);
+            var createListenerTask = binding.GetType().GetMethod("CreateListenerAsync");
+            var listener = await (Task<IListener>)createListenerTask.Invoke(binding, parameters);
+            var listenerOptions = (ServiceBusOptions)listener.GetType().GetField("_serviceBusOptions", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(listener);
+            var maxMessageBatchSizeSentToListener = (int)listener.GetType().GetField("_maxMessageBatchSize", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(listener);
+
+            Assert.NotNull(listenerOptions);
+            Assert.AreEqual(listenerOptions.MaxMessageBatchSize, 1000);
+            Assert.AreEqual(maxMessageBatchSizeSentToListener, 2);
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/Bindings/ServiceBusTriggerAttributeBindingProviderTests.cs:145]: Do we also need a negative test that verifies when the trigger is not set, the host.json option is used?

**Comment by JoshLove-msft** [comment]
> @apranaseth: We'll need to add a change log entry for this, and also to update the MS docs content for the `host.json` file to call out that the value may be overridden locally.

I will follow up with the change log entry when preparing for release. I forwarded on the details (thanks @jsquire!) to @apranaseth  for updating the msdocs page.

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**